cmd/bk/main.go

```diff
@@ -63,8 +89,52 @@ func run(args []string, exit func(int)) {
 		if debugGraphQL {
 			graphql.DebugHTTP = true
 		}
+
+		// infer keyring backend types if we get certain config
+		if keyringFileDir != "" {
+			keyringBackend = `file`
+		} else if keyringKeychain != "" {
+			keyringBackend = `keychain`
+		} else if keyringPassDir != "" {
+			keyringBackend = `pass`
+		}
+
+		var allowedBackends []keyring.BackendType
+		if keyringBackend != `` {
+			allowedBackends = append(allowedBackends, keyring.BackendType(keyringBackend))
+		}
+
+		// otherwise use one of the defaults
+		if keyringBackend == `` {
+			for _, k := range backendsAvailable {
+				switch keyring.BackendType(k) {
+				// secret-service and kwallet are kind of the worst
+				case keyring.KWalletBackend, keyring.SecretServiceBackend:
+					continue
+				default:
+					allowedBackends = append(allowedBackends, keyring.BackendType(k))
+				}
+			}
+		}
+
+		// default keychain to the login keychain
+		if keyringBackend == `keyring` && keyringKeychain == `` {
+			keyringKeychain = `login`
+		}
+
 		keyringImpl, err = keyring.Open(keyring.Config{
-			ServiceName: "buildkite",
+			ServiceName:              "buildkite",
+			AllowedBackends:          allowedBackends,
+			KeychainName:             keyringKeychain,
+			FileDir:                  "~/.buildkite/keyring/",
```

Comment:
Could that become `.config/buildkite/keyring` by default?

Complies with XDG directory specification.  https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html